flink-table-store-core/src/main/java/org/apache/flink/table/store/file/data/AppendOnlyWriter.java

```diff
@@ -92,22 +108,30 @@ public void write(RowData rowData) throws Exception {
 
     @Override
     public Increment prepareCommit() throws Exception {
-        List<DataFileMeta> newFiles = new ArrayList<>();
+        submitCompaction();
```

Comment:
Add new Files first, and then `submitCompaction`?

Reply:
> Add new Files first, and then submitCompaction?

Could you explain more details about why the order matters?